test/integration/routes/siti-agri/agreements.test.js

```diff
@@ -0,0 +1,32 @@
+import { retrieveOrganisationAgreements } from '../../../../src/factories/siti-agri/agreement.factory.js'
+
+describe('Basic queries for faked routes', () => {
+  let mockServer
+  const PROCESS_ENV = process.env
+
+  beforeAll(async () => {
+    process.env = { ...PROCESS_ENV }
+    process.env.PORT = '3097' // Set to obscure port to avoid conflicts
+    const { startServer } = await import('../../../../src/server.js')
+    mockServer = await startServer()
+  })
+  afterAll(() => {
+    process.env = PROCESS_ENV
+    mockServer.stop({ timeout: 0 })
+  })
```

Comment:
There's no need to start the full server here (those checks will be done in the contract tests) - instead you can keep the test lighter/faster:
```suggestion
  let server, schema
  beforeAll(async () => {
    server = Hapi.server()
    server.route(person)
    await Promise.all([
      server.initialize(),
      loadSchema('src/routes/v2/person-schema.yml').then((s) => (schema = s))
    ])
  })
```
This just `initialize`s the server which is enough to be able to `inject` requests.